src/js/node/stream.ts

```diff
@@ -4350,7 +4354,85 @@ var require_writable = __commonJS({
     Writable.prototype[EE.captureRejectionSymbol] = function (err) {
       this.destroy(err);
     };
-    var webStreamsAdapters;
+    var webStreamsAdapters = {
+      newStreamWritableFromWritableStream(
+        writableStream: WritableStream,
+        options: {
+          highWaterMark?: number;
+          encoding?: string;
+          objectMode?: boolean;
+          signal?: AbortSignal;
+        } = {},
+      ) {
+        if (!isWritableStream(writableStream)) {
+          throw new ERR_INVALID_ARG_TYPE("writableStream", "WritableStream", writableStream);
+        }
+
+        validateObject(options, "options");
+        const {
+          highWaterMark,
+          encoding,
+          objectMode = false,
+          signal,
+          // native = true,
+        } = options;
+
+        if (encoding !== undefined && !Buffer.isEncoding(encoding)) {
+          // TODO: why does this err in typescript?
+          throw new ERR_INVALID_ARG_VALUE(encoding, "options.encoding");
+        }
+        validateBoolean(objectMode, "options.objectMode");
+
+        const nativeStream = getNativeWritableStream(writableStream, options);
+        return nativeStream;
+      },
+
+      newWritableStreamFromStreamWritable(streamWritable: any, options: any = {}) {
+        if (typeof streamWritable?._writableState !== "object") {
+          throw new ERR_INVALID_ARG_TYPE("streamReadable", "stream.Readable", streamWritable);
+        }
+        var { isDestroyed, isWritable } = require_utils();
+        if (isDestroyed(streamWritable) || !isWritable(streamWritable)) {
+          const writable = new WritableStream();
+          writable.close();
+          return writable;
+        }
+        const objectMode = streamWritable.writableObjectMode;
+        const highWaterMark = streamWritable.writableHighWaterMark;
+
+        const evaluateStrategyOrFallback = strategy => {
```

Comment:
does this need to be a new function each time?

Reply:
probably not, but mostly copied over from `newStreamReadableFromReadableStream`, and not sure about performance or correctness changes as a result of extracting the function.